### DISCUSSION TITLE
ChatCompletion->create_async

### DISCUSSION THREAD

**Discussion by asnagni** [D7EAD/liboai#56] [Q&A] [answered]
Hi guys,
   Do you have anywhere in the documentation an example showcasing the **Chatcompletion** with an **asynchronus** call passing a **callback function**?
 
      **ChatCompletion->create_async**
   I couldn't find any example in the documentation. 
 
 The issue that I'm facing is that the AI agent will send me a reply in two or 3 parts (not in one shot). How can I make sure that I receive all data?

Thank you,
Stay safe

**Comment by pkusunjy** [comment, marked as answer]
Hi @asnagni ,

Perhaps you need to pass your own callback function to the `stream` parameter in `create_async`, and here is a tiny demo illustrating how it works. 
``` cpp
// define your own callback function
auto callback = [](std::string data, intptr_t ptr) -> bool {
    std::cout << data << '\n';
    return true;
};

// pass it to stream
xxx.create_async(model, conversation, std::nullopt, std::nullopt, std::nullopt, callback);
```
The stream will return streaming messages with `data: {somejsondata}` and be closed by a `data: [DONE]` message. 

Hope this may help you.

**Reply by asnagni** [reply]
Sorry for the delay and thank you for you answer, I do appreciate.


Stay safe
